validator/client/metrics.go

```diff
@@ -122,8 +122,8 @@ var (
 // and penalties over time, percentage gain/loss, and gives the end user a better idea
 // of how the validator performs with respect to the rest.
 func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64) error {
-	if slot%params.BeaconConfig().SlotsPerEpoch != 0 || slot <= params.BeaconConfig().SlotsPerEpoch {
-		// Do nothing unless we are at the start of the epoch, and not in the first epoch.
+	if (slot+1)%params.BeaconConfig().SlotsPerEpoch != 0 || slot <= params.BeaconConfig().SlotsPerEpoch {
```

Comment:
I think this should be at the start of the epoch. 

Example, at slot 64, it logs balances of epoch 1 that is slot 32 to 63. End of the epoch is not exactly accurate by definition

Reply:
The thing is the call to `GetValidatorPerformance`, the error if you call it before the attestation is included is a lot: it changes from missing to not missing. This affects the summary logged in the end of that function. 

As for the balances, the error if any is minimal (only for any change in the last slot), and is also irrelevant cause any difference in the balance will be included in the next measurement. But the error in inclusion_distance, voted head, etc. is never corrected since these are not cumulative.